bug: error after execute add column

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
execute the following sql in upstream database:
```
MySQL [test]> create table t1 (a int primary key );
Query OK, 0 rows affected (1.137 sec)

MySQL [test]> ALTER TABLE t1 ADD COLUMN c1 INT NOT NULL;
Query OK, 0 rows affected (2.360 sec)

MySQL [test]> insert into t1 value (1,2);
Query OK, 1 row affected (0.003 sec)
```
2. What did you expect to see?

3. What did you see instead?
error in cdc log:
```
[2020/03/25 16:08:09.488 +08:00] [ERROR] [scheduler.go:70] ["error on running processor"] [captureid=ef04810f-5fb3-4477-a08c-46b9d769fc12] [changefeedid=297a418f-2fc4-4e2c-a7a0-008b1d4e8189] [processorid=b8906575-30de-4880-b162-cab31fa176b6] [error="row commitTs: 415530663422197764: Error 1364: Field 'c1' doesn't have a default value"] [errorVerbose="Error 1364: Field 'c1' doesn't have a default value\nrow commitTs: 415530663422197764\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDMLs\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:449\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).concurrentExec.func1.1\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:353\ngithub.com/pingcap/ticdc/cdc/sink.rowLimitIterator\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:382\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).concurrentExec.func1\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:350\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1373"]
```
4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - TiCDC version (execute `cdc version`):

        ```
        (paste TiCDC version here)
        ```

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -126,13 +126,14 @@ func (m *mounterImpl) Run(ctx context.Context) error {
 			return errors.Trace(ctx.Err())
 		}
 
+		log.Info("processor HandlePreviousDDLJobIfNeed", zap.Uint64("ts", rawRow.Ts))
+		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts, true)
+
 		if rawRow.OpType == model.OpTypeResolved {
 			m.output <- &model.RowChangedEvent{Resolved: true, Ts: rawRow.Ts}
 			continue
 		}
 
-		log.Info("processor HandlePreviousDDLJobIfNeed", zap.Uint64("ts", rawRow.Ts))
-		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts, true)
 		switch errors.Cause(err) {
 		case nil:
 		case model.ErrUnresolved:

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -115,6 +115,9 @@ func (m *mounterImpl) Run(ctx context.Context) error {
 			return errors.Trace(ctx.Err())
 		case rawRow = <-m.rawRowChangedCh:
 		}
+		if rawRow == nil {
+			continue
+		}
 
 		if err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts); err != nil {
 			return errors.Trace(err)

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -196,6 +196,7 @@ func (m *mounterImpl) unmarshalAndMountRowChanged(raw *model.RawKVEntry) (*model
 func (m *mounterImpl) unmarshalRowKVEntry(restKey []byte, rawValue []byte, base baseKVEntry) (*rowKVEntry, error) {
 	tableID := base.TableID
 	tableInfo, exist := m.schemaStorage.TableByID(tableID)
+	log.Info("unmarshalRowKVEntry", zap.Reflect("tableInfo", tableInfo))
 	if !exist {
 		if m.schemaStorage.IsTruncateTableID(tableID) {
 			log.Debug("skip the DML of truncated table", zap.Uint64("ts", base.Ts), zap.Int64("tableID", tableID))
@@ -282,6 +283,7 @@ func (m *mounterImpl) mountRowKVEntry(row *rowKVEntry) (*model.RowChangedEvent, 
 	if !exist {
 		return nil, errors.NotFoundf("table in schema storage, id: %d", row.TableID)
 	}
+	log.Info("mountRowKVEntry", zap.Reflect("tableInfo", tableInfo), zap.Reflect("tableName", tableName))
 
 	if row.Delete && !tableInfo.PKIsHandle {
 		return nil, nil

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -126,8 +126,7 @@ func (m *mounterImpl) Run(ctx context.Context) error {
 			return errors.Trace(ctx.Err())
 		}
 
-		log.Info("processor HandlePreviousDDLJobIfNeed", zap.Uint64("ts", rawRow.Ts))
-		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts, true)
+		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts)
 
 		if rawRow.OpType == model.OpTypeResolved {
 			m.output <- &model.RowChangedEvent{Resolved: true, Ts: rawRow.Ts}
@@ -198,7 +197,6 @@ func (m *mounterImpl) unmarshalAndMountRowChanged(raw *model.RawKVEntry) (*model
 func (m *mounterImpl) unmarshalRowKVEntry(restKey []byte, rawValue []byte, base baseKVEntry) (*rowKVEntry, error) {
 	tableID := base.TableID
 	tableInfo, exist := m.schemaStorage.TableByID(tableID)
-	log.Info("unmarshalRowKVEntry", zap.Reflect("tableInfo", tableInfo))
 	if !exist {
 		if m.schemaStorage.IsTruncateTableID(tableID) {
 			log.Debug("skip the DML of truncated table", zap.Uint64("ts", base.Ts), zap.Int64("tableID", tableID))
@@ -285,7 +283,6 @@ func (m *mounterImpl) mountRowKVEntry(row *rowKVEntry) (*model.RowChangedEvent, 
 	if !exist {
 		return nil, errors.NotFoundf("table in schema storage, id: %d", row.TableID)
 	}
-	log.Info("mountRowKVEntry", zap.Reflect("tableInfo", tableInfo), zap.Reflect("tableName", tableName))
 
 	if row.Delete && !tableInfo.PKIsHandle {
 		return nil, nil

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -131,7 +131,8 @@ func (m *mounterImpl) Run(ctx context.Context) error {
 			continue
 		}
 
-		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts)
+		log.Info("processor HandlePreviousDDLJobIfNeed", zap.Uint64("ts", rawRow.Ts))
+		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts, true)
 		switch errors.Cause(err) {
 		case nil:
 		case model.ErrUnresolved:

--- a/cdc/entry/mounter.go
+++ b/cdc/entry/mounter.go
@@ -19,7 +19,6 @@ import (
 	"encoding/binary"
 	"encoding/json"
 	"fmt"
-	"time"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
@@ -109,38 +108,21 @@ func NewMounter(rawRowChangedCh <-chan *model.RawKVEntry, schemaStorage *Storage
 }
 
 func (m *mounterImpl) Run(ctx context.Context) error {
-	var lastRowChangedEvent *model.RawKVEntry
 	for {
 		var rawRow *model.RawKVEntry
-		if lastRowChangedEvent != nil {
-			rawRow = lastRowChangedEvent
-			lastRowChangedEvent = nil
-		} else {
-			select {
-			case <-ctx.Done():
-				return errors.Trace(ctx.Err())
-			case rawRow = <-m.rawRowChangedCh:
-			}
-		}
-		if rawRow == nil {
+		select {
+		case <-ctx.Done():
 			return errors.Trace(ctx.Err())
+		case rawRow = <-m.rawRowChangedCh:
 		}
 
-		err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts)
+		if err := m.schemaStorage.HandlePreviousDDLJobIfNeed(rawRow.Ts); err != nil {
+			return errors.Trace(err)
+		}
 
 		if rawRow.OpType == model.OpTypeResolved {
 			m.output <- &model.RowChangedEvent{Resolved: true, Ts: rawRow.Ts}
 			continue
-		}
-
-		switch errors.Cause(err) {
-		case nil:
-		case model.ErrUnresolved:
-			lastRowChangedEvent = rawRow
-			time.Sleep(50 * time.Millisecond)
-			continue
-		default:
-			return errors.Cause(err)
 		}
 
 		event, err := m.unmarshalAndMountRowChanged(rawRow)

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -196,10 +196,10 @@ func (b *StorageBuilder) DoGc(ts uint64) error {
 	}
 	b.baseStorageMu.Lock()
 	defer b.baseStorageMu.Unlock()
-	//err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
-	//if err != nil {
-	//	return errors.Trace(err)
-	//}
+	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
+	if err != nil {
+		return errors.Trace(err)
+	}
 	//b.jobList.RemoveOverdueJobs(ts)
 	return nil
 }

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -196,10 +196,10 @@ func (b *StorageBuilder) DoGc(ts uint64) error {
 	}
 	b.baseStorageMu.Lock()
 	defer b.baseStorageMu.Unlock()
-	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	b.jobList.RemoveOverdueJobs(ts)
+	//err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
+	//if err != nil {
+	//	return errors.Trace(err)
+	//}
+	//b.jobList.RemoveOverdueJobs(ts)
 	return nil
 }

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -38,12 +38,13 @@ func (l *jobList) FetchNextJobs(currentJob *list.Element, ts uint64) (*list.Elem
 	l.mu.RLock()
 	defer l.mu.RUnlock()
 
-	if ts < l.gcTs {
+	if ts <= l.gcTs {
 		log.Fatal("cannot fetch the jobs which of finishedTs is less then gcTs, please report a bug", zap.Uint64("gcTs", l.gcTs))
 	}
 
 	if currentJob != l.list.Front() {
 		job := currentJob.Value.(*timodel.Job)
+		// check if element was already gced.
 		if job.BinlogInfo.FinishedTS <= l.gcTs {
 			currentJob = l.list.Front()
 		}
@@ -65,7 +66,7 @@ func (l *jobList) AppendJob(jobs ...*timodel.Job) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	for _, job := range jobs {
-		if job.BinlogInfo.FinishedTS < l.gcTs {
+		if job.BinlogInfo.FinishedTS <= l.gcTs {
 			log.Fatal("cannot append a job which of finishedTs is less then gcTs, please report a bug", zap.Uint64("gcTs", l.gcTs), zap.Reflect("job", job))
 		}
 		l.list.PushBack(job)
@@ -75,14 +76,16 @@ func (l *jobList) AppendJob(jobs ...*timodel.Job) {
 func (l *jobList) RemoveOverdueJobs(ts uint64) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
+	log.Info("BUGFOUNDER gcts", zap.Uint64("gcts", ts))
 	for e := l.list.Front().Next(); e != nil; {
 		job := e.Value.(*timodel.Job)
-		if job.BinlogInfo.FinishedTS > ts {
+		if job.BinlogInfo.FinishedTS >= ts {
 			break
 		}
 		l.gcTs = job.BinlogInfo.FinishedTS
 		lastE := e
 		e = e.Next()
+		log.Info("BUGFOUNDER gc job", zap.Reflect("job", lastE.Value))
 		l.list.Remove(lastE)
 	}
 }
@@ -200,6 +203,6 @@ func (b *StorageBuilder) DoGc(ts uint64) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	//b.jobList.RemoveOverdueJobs(ts)
+	b.jobList.RemoveOverdueJobs(ts)
 	return nil
 }

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -170,7 +170,7 @@ func (b *StorageBuilder) Build(ts uint64) (*Storage, error) {
 	b.baseStorageMu.Unlock()
 
 	err := retry.Run(func() error {
-		err := c.HandlePreviousDDLJobIfNeed(ts)
+		err := c.HandlePreviousDDLJobIfNeed(ts, false)
 		if errors.Cause(err) != model.ErrUnresolved {
 			return backoff.Permanent(err)
 		}
@@ -196,7 +196,7 @@ func (b *StorageBuilder) DoGc(ts uint64) error {
 	}
 	b.baseStorageMu.Lock()
 	defer b.baseStorageMu.Unlock()
-	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts)
+	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
 	if err != nil {
 		return errors.Trace(err)
 	}

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -34,6 +34,9 @@ func (l *jobList) FetchNextJobs(currentJob *list.Element, ts uint64) (*list.Elem
 	if currentJob == nil {
 		log.Fatal("param `currentJob` in `FetchNextJobs` can't be nil, please report a bug")
 	}
+	if ts == 0 {
+		return currentJob, nil
+	}
 
 	l.mu.RLock()
 	defer l.mu.RUnlock()
@@ -76,7 +79,6 @@ func (l *jobList) AppendJob(jobs ...*timodel.Job) {
 func (l *jobList) RemoveOverdueJobs(ts uint64) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
-	log.Info("BUGFOUNDER gcts", zap.Uint64("gcts", ts))
 	for e := l.list.Front().Next(); e != nil; {
 		job := e.Value.(*timodel.Job)
 		if job.BinlogInfo.FinishedTS >= ts {
@@ -85,7 +87,6 @@ func (l *jobList) RemoveOverdueJobs(ts uint64) {
 		l.gcTs = job.BinlogInfo.FinishedTS
 		lastE := e
 		e = e.Next()
-		log.Info("BUGFOUNDER gc job", zap.Reflect("job", lastE.Value))
 		l.list.Remove(lastE)
 	}
 }
@@ -173,7 +174,7 @@ func (b *StorageBuilder) Build(ts uint64) (*Storage, error) {
 	b.baseStorageMu.Unlock()
 
 	err := retry.Run(func() error {
-		err := c.HandlePreviousDDLJobIfNeed(ts, false)
+		err := c.HandlePreviousDDLJobIfNeed(ts)
 		if errors.Cause(err) != model.ErrUnresolved {
 			return backoff.Permanent(err)
 		}
@@ -199,7 +200,7 @@ func (b *StorageBuilder) DoGc(ts uint64) error {
 	}
 	b.baseStorageMu.Lock()
 	defer b.baseStorageMu.Unlock()
-	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts, false)
+	err := b.baseStorage.HandlePreviousDDLJobIfNeed(ts)
 	if err != nil {
 		return errors.Trace(err)
 	}

--- a/cdc/entry/schema_builder.go
+++ b/cdc/entry/schema_builder.go
@@ -7,12 +7,10 @@ import (
 	"sync"
 	"sync/atomic"
 
-	"github.com/cenkalti/backoff"
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	timodel "github.com/pingcap/parser/model"
 	"github.com/pingcap/ticdc/cdc/model"
-	"github.com/pingcap/ticdc/pkg/retry"
 	"go.uber.org/zap"
 )
 
@@ -173,14 +171,7 @@ func (b *StorageBuilder) Build(ts uint64) (*Storage, error) {
 	c := b.baseStorage.Clone()
 	b.baseStorageMu.Unlock()
 
-	err := retry.Run(func() error {
-		err := c.HandlePreviousDDLJobIfNeed(ts)
-		if errors.Cause(err) != model.ErrUnresolved {
-			return backoff.Permanent(err)
-		}
-		return err
-	}, 5)
-	if err != nil {
+	if err := c.HandlePreviousDDLJobIfNeed(ts); err != nil {
 		return nil, errors.Trace(err)
 	}
 	return c, nil

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -604,7 +604,7 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		if binlogInfo == nil {
 			return "", "", "", errors.NotFoundf("table %d", job.TableID)
 		}
-		tbInfo := binlogInfo.TableInfo
+		tbInfo := binlogInfo.TableInfo.Clone()
 		if tbInfo == nil {
 			return "", "", "", errors.NotFoundf("table %d", job.TableID)
 		}

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -439,6 +439,9 @@ func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64, processor bool) er
 	}
 	currentJob, jobs := s.jobList.FetchNextJobs(s.currentJob, commitTs)
 	for _, job := range jobs {
+		if processor {
+			log.Info("jobs in HandlePreviousDDLJobIfNeed processor", zap.Reflect("job", job))
+		}
 		if SkipJob(job) {
 			log.Info("skip DDL job because the job isn't synced and done", zap.Stringer("job", job))
 			continue

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -397,7 +397,6 @@ func (s *Storage) CreateTable(schema *timodel.DBInfo, table *timodel.TableInfo) 
 	schema.Tables = append(schema.Tables, table)
 	s.tables[table.ID] = WrapTableInfo(table)
 	s.tableIDToName[table.ID] = TableName{Schema: schema.Name.O, Table: table.Name.O}
-	log.Info("create table in schema", zap.String("table", table.Name.O))
 	s.tableNameToID[s.tableIDToName[table.ID]] = table.ID
 
 	log.Debug("create table success", zap.String("name", schema.Name.O+"."+table.Name.O), zap.Int64("id", table.ID))
@@ -433,15 +432,12 @@ func (s *Storage) removeTable(tableID int64) error {
 }
 
 // HandlePreviousDDLJobIfNeed apply all jobs with FinishedTS less or equals `commitTs`.
-func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64, processor bool) error {
+func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64) error {
 	if commitTs > atomic.LoadUint64(s.resolvedTs) {
 		return model.ErrUnresolved
 	}
 	currentJob, jobs := s.jobList.FetchNextJobs(s.currentJob, commitTs)
 	for _, job := range jobs {
-		if processor {
-			log.Info("jobs in HandlePreviousDDLJobIfNeed processor", zap.Reflect("job", job))
-		}
 		if SkipJob(job) {
 			log.Info("skip DDL job because the job isn't synced and done", zap.Stringer("job", job))
 			continue
@@ -449,9 +445,6 @@ func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64, processor bool) er
 		if job.BinlogInfo.FinishedTS <= s.lastHandledTs {
 			log.Debug("skip DDL job because the job is already handled", zap.Stringer("job", job))
 			continue
-		}
-		if processor {
-			log.Info("handle ", zap.Reflect("job", job))
 		}
 		_, _, _, err := s.HandleDDL(job)
 		if err != nil {
@@ -527,7 +520,6 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		}
 		// create table
 		table := job.BinlogInfo.TableInfo.Clone()
-		log.Info("rename table in schema ", zap.Reflect("new table", table))
 		schema, ok := s.SchemaByID(job.SchemaID)
 		if !ok {
 			return "", "", "", errors.NotFoundf("schema %d", job.SchemaID)

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -19,6 +19,9 @@ import (
 	"fmt"
 	"sync/atomic"
 
+	"github.com/cenkalti/backoff"
+	"github.com/pingcap/ticdc/pkg/retry"
+
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	timodel "github.com/pingcap/parser/model"
@@ -433,6 +436,22 @@ func (s *Storage) removeTable(tableID int64) error {
 
 // HandlePreviousDDLJobIfNeed apply all jobs with FinishedTS less or equals `commitTs`.
 func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64) error {
+	err := retry.Run(func() error {
+		err := s.handlePreviousDDLJobIfNeed(commitTs)
+		if errors.Cause(err) != model.ErrUnresolved {
+			return backoff.Permanent(err)
+		}
+		return err
+	}, 5)
+	switch err.(type) {
+	case *backoff.PermanentError:
+		return errors.New("timeout when waiting resolved ts of ddl puller")
+	default:
+		return err
+	}
+}
+
+func (s *Storage) handlePreviousDDLJobIfNeed(commitTs uint64) error {
 	if commitTs > atomic.LoadUint64(s.resolvedTs) {
 		return model.ErrUnresolved
 	}

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -433,7 +433,7 @@ func (s *Storage) removeTable(tableID int64) error {
 }
 
 // HandlePreviousDDLJobIfNeed apply all jobs with FinishedTS less or equals `commitTs`.
-func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64) error {
+func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64, processor bool) error {
 	if commitTs > atomic.LoadUint64(s.resolvedTs) {
 		return model.ErrUnresolved
 	}
@@ -446,6 +446,9 @@ func (s *Storage) HandlePreviousDDLJobIfNeed(commitTs uint64) error {
 		if job.BinlogInfo.FinishedTS <= s.lastHandledTs {
 			log.Debug("skip DDL job because the job is already handled", zap.Stringer("job", job))
 			continue
+		}
+		if processor {
+			log.Info("handle ", zap.Reflect("job", job))
 		}
 		_, _, _, err := s.HandleDDL(job)
 		if err != nil {

--- a/cdc/entry/schema_storage.go
+++ b/cdc/entry/schema_storage.go
@@ -397,6 +397,7 @@ func (s *Storage) CreateTable(schema *timodel.DBInfo, table *timodel.TableInfo) 
 	schema.Tables = append(schema.Tables, table)
 	s.tables[table.ID] = WrapTableInfo(table)
 	s.tableIDToName[table.ID] = TableName{Schema: schema.Name.O, Table: table.Name.O}
+	log.Info("create table in schema", zap.String("table", table.Name.O))
 	s.tableNameToID[s.tableIDToName[table.ID]] = table.ID
 
 	log.Debug("create table success", zap.String("name", schema.Name.O+"."+table.Name.O), zap.Int64("id", table.ID))
@@ -487,7 +488,7 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		schemaName = schema.Name.O
 
 	case timodel.ActionModifySchemaCharsetAndCollate:
-		db := job.BinlogInfo.DBInfo
+		db := job.BinlogInfo.DBInfo.Clone()
 		if _, ok := s.schemas[db.ID]; !ok {
 			return "", "", "", errors.NotFoundf("schema %s(%d)", db.Name, db.ID)
 		}
@@ -508,7 +509,7 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		s.currentVersion = job.BinlogInfo.SchemaVersion
 
 	case timodel.ActionRenameTable:
-		// ignore schema doesn't support reanme ddl
+		// ignore schema doesn't support rename ddl
 		_, ok := s.SchemaByTableID(job.TableID)
 		if !ok {
 			return "", "", "", errors.NotFoundf("table(%d) or it's schema", job.TableID)
@@ -520,6 +521,7 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		}
 		// create table
 		table := job.BinlogInfo.TableInfo.Clone()
+		log.Info("rename table in schema ", zap.Reflect("new table", table))
 		schema, ok := s.SchemaByID(job.SchemaID)
 		if !ok {
 			return "", "", "", errors.NotFoundf("schema %d", job.SchemaID)
@@ -604,10 +606,11 @@ func (s *Storage) HandleDDL(job *timodel.Job) (schemaName string, tableName stri
 		if binlogInfo == nil {
 			return "", "", "", errors.NotFoundf("table %d", job.TableID)
 		}
-		tbInfo := binlogInfo.TableInfo.Clone()
+		tbInfo := binlogInfo.TableInfo
 		if tbInfo == nil {
 			return "", "", "", errors.NotFoundf("table %d", job.TableID)
 		}
+		tbInfo = tbInfo.Clone()
 
 		schema, ok := s.SchemaByID(job.SchemaID)
 		if !ok {

--- a/cdc/model/errors.go
+++ b/cdc/model/errors.go
@@ -29,5 +29,5 @@ var (
 	ErrAdminStopProcessor     = errors.New("stop processor by admin command")
 	ErrExecDDLFailed          = errors.New("exec DDL failed")
 	ErrCaptureNotExist        = errors.New("capture not exists")
-	ErrUnresolved             = errors.New("the ts more than resolvedTs")
+	ErrUnresolved             = errors.New("unresolved")
 )

--- a/cdc/processor.go
+++ b/cdc/processor.go
@@ -600,8 +600,8 @@ func (p *processor) addTable(ctx context.Context, tableID int64, startTs uint64)
 	}()
 	storage, err := p.schemaBuilder.Build(startTs)
 	if err != nil {
-		log.Error("failed to build schema storage", zap.Error(err))
 		p.errCh <- errors.Trace(err)
+		return
 	}
 	// start mounter
 	mounter := entry.NewMounter(puller.SortedOutput(ctx), storage)

--- a/cdc/processor.go
+++ b/cdc/processor.go
@@ -600,6 +600,7 @@ func (p *processor) addTable(ctx context.Context, tableID int64, startTs uint64)
 	}()
 	storage, err := p.schemaBuilder.Build(startTs)
 	if err != nil {
+		log.Error("failed to build schema storage", zap.Error(err))
 		p.errCh <- errors.Trace(err)
 	}
 	// start mounter

